getter.go

```diff
@@ -153,6 +158,11 @@ func (g *getter) initChunks() {
 func (g *getter) worker() {
 	for c := range g.getCh {
 		g.retryGetChunk(c)
+		if g.err != nil {
+			// tell Read() caller that 1 or more chunks can't be read; abort
+			g.abortOnce.Do(func() { close(g.workerAborted) })
```

Comment:
Oh, that's nice. I just did something similar in a PR [here](https://github.com/github/spokesd/pull/943/files#diff-595c6f3ea9cb61a6a8813c0d7665c8a1f71c1b1de6235eec6fa5af2abb33537eR598-R603). This looks safer